department-of-veterans-affairs/caseflow
# CAMO sends to BVA Intake

Resolves [CASEFLOW-2181](https://vajira.max.gov/browse/CASEFLOW-2181)

### Description
CAMO can complete task and return to BVA Intake. Note, I don't think there are designs for this right now, so this is a placeholder.

### Acceptance Criteria
- [x] A user of the CAMO organization has an available action to complete their task
- [x] Selecting to complete the task shows a modal with a box to enter instructions
- [x] The CAMO user can enter notes, which are added to the instructions on the parent BVA Intake task
- [x] The CAMO task is marked as completed, and the parent BVA Intake task moves from on hold to assigned

### Testing Plan
1. Go to Intake as a BVAIntake user
2. Go through the intake process creating a pre-docket appeal (appeal with a VHA issue)
3. Sign in as a CAMO user
4. Visit the CAMO org queue and visit the appeal that was just created
5. Make sure the action to "Send to Board Intake" is available
6. Select the action
7. Compete the modal
8. Ensure the success banner appears when you are redirected back to the queue 
9. Sign is as the BVA Intake user once again
10. Go to the BVA Intake org queue and select the appeal
11. Ensure that there are available actions for that appeal (confirming their `PreDocketTask` has moved from `on_hold` to `assigned`)

### User Facing Changes
#### CAMO pre docket available actions
![Screen Shot 2021-09-28 at 2 49 00 PM](https://user-images.githubusercontent.com/18618189/135148232-4f42549f-7915-4c08-828c-89e2618d793e.png)
#### CAMO confirmation modal
![Screen Shot 2021-09-28 at 2 49 14 PM](https://user-images.githubusercontent.com/18618189/135148246-4c6fe2fd-b81f-4900-8b0b-611380c94939.png)
#### CAMO success banner
![Screen Shot 2021-09-28 at 2 49 44 PM](https://user-images.githubusercontent.com/18618189/135148261-7a5961a6-2bab-4b0a-9c66-bd7acb932910.png)
#### BVA Intake pre docket available actions (showing appeal is now assigned to BVA)
![Screen Shot 2021-09-28 at 2 52 13 PM](https://user-images.githubusercontent.com/18618189/135148267-649cf059-cbc9-480e-84c2-0d1a15d91298.png)
#### BVA Intake task instructions in Case Timeline
![Screen Shot 2021-09-28 at 3 12 13 PM](https://user-images.githubusercontent.com/18618189/135151008-08bbdb13-b02d-47b6-8370-1e4da7d1065b.png)
